src/base/bittorrent/session.cpp

```diff
@@ -1962,8 +1995,11 @@ bool Session::cancelLoadMetadata(const InfoHash &hash)
     m_loadedMetadata.remove(hash);
     libt::torrent_handle torrent = m_nativeSession->find_torrent(hash);
     if (!torrent.is_valid()) return false;
-
+#if LIBTORRENT_VERSION_NUM < 10200
     if (!torrent.status(0).has_metadata) {
+#else
+    if (!torrent.status(libt::status_flags_t(0)).has_metadata) {
```

Comment:
instead of `status_flags_t(0)`, I'm pretty sure you can just say `{}`, (in fact, that might work in both libtorrent 1.1.x and 1.2.x).

However, keep in mind that a call to `status()` is quite expensive, since it's a round-trip to the libtorrent thread.

Reply:
>However, keep in mind that a call to status() is quite expensive, since it's a round-trip to the libtorrent thread.

We try to use it only when it is impossible to use cached values.